### DISCUSSION TITLE
Building multimodal systems

### DISCUSSION THREAD

**Discussion by AmirHussein96** [k2-fsa/icefall#1536] [Ideas]
How difficult is it to develop a multimodal system that integrates text, speech, and images using Icefall? Given that Icefall and lhotse currently support speech and text, how difficult would it be to extend its capabilities to include images?

**Comment by JinZr** [comment]
Shouldn't be too hard as Lhotse has recently supported dataloading to text/multimodal data.

Check https://github.com/lhotse-speech/lhotse/pull/1295 for more details.

**Reply by AmirHussein96** [reply]
Thanks
